packages/slate/src/interfaces/editor.ts

```diff
@@ -1736,3 +1736,4 @@ export type NodeMatch<T extends Node> =
   | ((node: Node, path: Path) => boolean)
 
 export type PropsCompare = (prop: Partial<Node>, node: Partial<Node>) => boolean
+export type PropsMerge = (prop: Partial<Node>, node: Partial<Node>) => object
```

Comment:
If I'm not mistaken, isn't this TS def (and the one for Compare) incorrect? It should be expecting a single property rather than a partial object of properties. And it won't necessarily get an object back, it could get anything.

A slightly more accurate version would be
```ts
type PropsMerge<T = unknown> = (prop: T | undefined, node: T) => T
```

Reply:
`PropsCompare` has the same bug FWIW.